src/remote_adapter/remote_adapter_client/qt_impl/sdl_remote_adapter_qt_client.cc

```diff
@@ -19,56 +28,35 @@ SDLRemoteTestAdapterQtClient::~SDLRemoteTestAdapterQtClient() {
     }
 }
 
-// void SDLRemoteTestAdapterQtClient::connect(const std::string& host, uint32_t port, const std::string& name) {
-//     name_ = name;
-//     try {
-//         remote_adapter_client_ptr_ = new SDLRemoteTestAdapterClient(host, port);
-//         if (0 == open(name)) {
-//             emit connected();
-//             // ToDo (aderiabin): Add incoming message listening
-//         } else {
-//             emit disconnected();
-//         }
-//     } catch (std::exception& e) {
-//         remote_adapter_client_ptr_ = nullptr;
-//         std::cout << e.what() << std::endl;
-//         emit disconnected();
-//     }
-// }
-
-void SDLRemoteTestAdapterQtClient::connectMq(const std::string& host,
-                                             uint32_t port,
-                                             MqParams& in_params,
-                                             MqParams& out_params) {
-    try {
-        remote_adapter_client_ptr_ = new SDLRemoteTestAdapterClient(host, port);
-        int in_open_result = openWithParams(in_params);
-        int out_open_result = openWithParams(out_params);
-        if ((0 == in_open_result) && (0 == out_open_result)) {
-            in_mq_name_ = in_params.name;
-            out_mq_name_ = out_params.name;
-
-            listener_ptr_ = new SDLRemoteTestAdapterReceiveThread(this);
-            connect(listener_ptr_, &SDLRemoteTestAdapterReceiveThread::dataAvailable,
-                    this, &SDLRemoteTestAdapterQtClient::textMessageReceived);
-            connect(listener_ptr_, &SDLRemoteTestAdapterReceiveThread::sourceUnreachable,
-                    this, &SDLRemoteTestAdapterQtClient::connectionLost);
-
-            isconnected_ = true;
-            emit connected();
-            listener_ptr_->start();
-        // } else {
-        //     // emit disconnected();
+void SDLRemoteTestAdapterQtClient::connectMq() {
+    if (!isconnected_) {
+        try {
+            remote_adapter_client_ptr_ = new SDLRemoteTestAdapterClient(host_, port_);
+            int in_open_result = openWithParams(in_mq_params_);
+            int out_open_result = openWithParams(out_mq_params_);
+            if ((0 == in_open_result) && (0 == out_open_result)) {
```

Comment:
@aderiabin  Maybe the situation when you get into this state twice? If so, you may still have a memory leak.

Reply:
@aderiabin We can use shared or unique pointer
instead of simple pointers ?

Reply:
@aderiabin use std::unique_ptr for remote_adapter_client_ptr_ and for listener_ptr_

Reply:
@SNiukalov @EKuliiev @BSolonenko `std::unique_ptr` is used for pointers on SDLRemoteTestAdapterClient and SDLRemoteTestAdapterReceiveThread instances 
(see bc3f86577b740bc84201d7d0edaf82cce1d8ec52)